pkg/infra/game/stage/loot/loot.go

```diff
@@ -157,3 +166,99 @@ func itemListToSet(
 	}
 	return res
 }
+
+func getAverageStats(globalState state.State) (float64, float64, float64, float64) {
+	var averageHP float64 = 0
+	var averageST float64 = 0
+	var averageATT float64 = 0
+	var averageDEF float64 = 0
+
+	agentLen := float64(len(globalState.AgentState))
+	for _, state := range globalState.AgentState {
+		averageHP += float64(state.Hp)
+		averageST += float64(state.Stamina)
+		averageATT += float64(state.BonusAttack())
+		averageDEF += float64(state.BonusDefense())
+	}
+
+	averageHP /= agentLen
+	averageST /= agentLen
+	averageATT /= agentLen
+	averageDEF /= agentLen
+	// fmt.Println(averageHP, averageST, averageATT, averageDEF)
+	meanAverageHP, meanAverageST, meanAverageATT, meanAverageDEF := normalize4El(averageHP, averageST, averageATT, averageDEF)
+	// fmt.Println(meanAverageHP, meanAverageST, meanAverageATT, meanAverageDEF)
+	return meanAverageHP, meanAverageST, meanAverageATT, meanAverageDEF
+}
+
+func chooseItem(agent state.AgentState, averageHP float64, averageST float64, averageATT float64, averageDEF float64) (bool, bool, bool, bool) {
+
+	HP := float64(agent.Hp)
+	ST := float64(agent.Stamina)
+	ATT := float64(agent.BonusAttack())
+	DEF := float64(agent.BonusDefense())
+	// fmt.Println(HP, ST, ATT, DEF)
+	meanHP, meanST, meanATT, meanDEF := normalize4El(HP, ST, ATT, DEF)
+	// fmt.Println(meanHP, meanST, meanATT, meanDEF)
+	diffHP := averageHP - meanHP
+	diffST := averageST - meanST
+	diffATT := averageATT - meanATT
+	diffDEF := averageDEF - meanDEF
+	// fmt.Println(diffHP, diffST, diffATT, diffDEF)
+	// get largest diff = var most in need
+	if diffHP > diffST && diffHP > diffATT && diffHP > diffDEF {
+		return true, false, false, false // HP highest diff
+	} else if diffST > diffATT && diffST > diffDEF {
+		return false, true, false, false // ST highest diff
+	} else if diffATT > diffDEF {
+		return false, false, true, false // ATT highest diff
+	} else {
+		return false, false, false, true // DEF highest diff
+	}
+
+}
+
+// works bc normalization changes the data distribution, so small sheild/weapon difference values are significant enough now
+func normalize4El(x, y, z, w float64) (float64, float64, float64, float64) {
+	maxVal := max4El(x, y, z, w)
+	minVal := min4El(x, y, z, w)
+	return (x - minVal) / (maxVal - minVal), (y - minVal) / (maxVal - minVal), (z - minVal) / (maxVal - minVal), (w - minVal) / (maxVal - minVal)
+}
+
+func max4El(x, y, z, w float64) float64 {
+	if x > y && x > z && x > w {
+		return x
+	}
+	if y > z && y > w {
+		return y
+	}
+	if z > w {
+		return z
+	}
+	return w
+}
+
+func min4El(x, y, z, w float64) float64 {
+	if x < y && x < z && x < w {
+		return x
+	}
+	if y < z && y < w {
+		return y
+	}
+	if z < w {
+		return z
+	}
+	return w
+}
```

Comment:
```
func minMax4(isMax bool, nums ...float64) float64 {
	ans := nums[0]
	for _, num := range nums[1:] {
		if isMax {
			ans = math.Max(num, ans)
		} else {
			ans = math.Min(num, ans)
		}
	}
	return ans
}
```

Reply:
you can combine these 2 functions easily (and still call in the same way, albeit with a bool at the start)

Reply:
will change this